src/coreclr/vm/jitinterface.cpp

```diff
@@ -6149,20 +6149,19 @@ bool CEEInfo::isObjectImmutable(CORINFO_OBJECT_HANDLE objHandle)
 
     _ASSERT(objHandle != NULL);
 
-#ifdef DEBUG
+    bool isImmutable = false;
+
     JIT_TO_EE_TRANSITION();
 
     GCX_COOP();
     OBJECTREF obj = getObjectFromJitHandle(objHandle);
     MethodTable* type = obj->GetMethodTable();
 
-    _ASSERTE(type->IsString() || type == g_pRuntimeTypeClass);
+    isImmutable = type->IsString() || type == g_pRuntimeTypeClass || type->GetNumInstanceFields() == 0;
```

Comment:
Does this change actually help with anything or is this just future proofing?

Reply:
Reverted, it's from my local branch where I experimented with `static readonly` + FOH 🙂